app/models/miq_report/generator.rb

```diff
@@ -244,7 +244,7 @@ def _generate_table(options = {})
       time_range = Metric::Helper.time_range_from_offset(interval, db_options[:start_offset], db_options[:end_offset])
 
       # Only build where clause from expression for hourly report. It will not work properly for daily because many values are rolled up from hourly.
-      exp_sql, exp_includes = conditions.to_sql(tz) unless conditions.nil? || klass.respond_to?(:instances_are_derived?)
+      exp_sql, exp_includes = conditions.to_sql unless conditions.nil? || klass.respond_to?(:instances_are_derived?)
```

Comment:
This is probably a tangent / scope creep:

I'm not a fan of `to_sql`, what do you think about:

```ruby
results = conditions.enhance(results)
```